File: src/src_core/fmg/core/img/PolarLightsImg.java
package fmg.core.img;

import fmg.common.HSV;

/**
 * {@link RotatedImg} with hue rotation effect.
 * Alters the color of an image by rotating its hue values.
 *
 * @param <TImage> plaform specific image
 */
public abstract class PolarLightsImg<TImage> extends RotatedImg<TImage> {

//   protected PolarLightsImg() { super(); }

   public static final String PROPERTY_POLAR_LIGHTS = "PolarLights";

   private boolean _polarLights;
   /** shimmering filling */
   public boolean isPolarLights() { return _polarLights; }
   public void setPolarLights(boolean value) {
      if (setProperty(_polarLights, value, PROPERTY_POLAR_LIGHTS)) {
         if (value)
            startTimer();
         else
            stopTimer();
      }
   }

   private void nextForegroundColor() {
      HSV hsv = new HSV(getForegroundColor());
      hsv.h += getRotateAngleDelta();
      setForegroundColor(hsv.toColor());
   }

   @Override
   protected void onTimer() {
      if (isPolarLights())
         nextForegroundColor();
      super.onTimer();
   }

   @Override
   public boolean isLiveImage() {
      return isPolarLights() || super.isLiveImage();
   }

}

File: src/src_core/fmg/core/img/PolarLightsImg.cs
using fmg.common;

namespace fmg.core.img {

   /// <summary> <see cref="RotatedImg{TImage}"/> with hue rotation effect.
   /// Alters the color of an image by rotating its hue values.
   /// </summary>
   /// <typeparam name="TImage">plaform specific image</typeparam>
   public abstract class PolarLightsImg<TImage> : RotatedImg<TImage>
      where TImage : class
   {
      //protected PolarLightsImg() : base() { }

      private bool _polarLights;
      /// <summary> shimmering filling </summary>
      public bool PolarLights {
         get { return _polarLights; }
         set {
            if (SetProperty(ref _polarLights, value))
               if (value)
                  StartTimer();
               else
                  StopTimer();
         }
      }

      private void NextForegroundColor() {
         var hsv = new HSV(ForegroundColor);
         hsv.h += RotateAngleDelta;
         ForegroundColor = hsv.ToColor();
      }

      protected override void OnTimer() {
         if (PolarLights)
            NextForegroundColor();
         base.OnTimer();
      }

      protected override bool LiveImage() {
         return PolarLights || base.LiveImage();
      }

   }
}
